docs/source/usersguide/geometry.rst

```diff
@@ -0,0 +1,353 @@
+- The pitch of the lattice, i.e., the distance between the center of adjacent
+  lattice elements (:attr:`RectLattice.pitch`),
+- What universes should fill each lattice element
+  (:attr:`RectLattice.universes`), and
+- A universe that is used to fill any lattice position outside the well-defined
+  portion of the lattice (:attr:`RectLattice.outer`).
+
+For example, to create a 3x3 lattice centered at the origin in which each
+lattice element is 5cm by 5cm and is filled by a universe ``u``, one could run::
+
+  lattice = openmc.RectLattice()
+  lattice.lower_left = (-7.5, -7.5)
+  lattice.pitch = (5.0, 5.0)
+  lattice.universes = [[u, u, u],
+                       [u, u, u],
+                       [u, u, u]]
+
+Note that because this is a two-dimensional lattice, the lower-left coordinates
+and pitch only need to specify the :math:`x,y` values. The order that the
+universes appear is such that the first row corresponds to lattice elements with
+the highest y-value. Note that the :attr:`RectLattice.universes` attribute
+expects a doubly-nested iterable of type :class:`openmc.Universe` --- this can
+be normal Python lists, as shown above, or a NumPy array can be used as well::
+
+  lattice.universes = np.tile(u, (3, 3))
+
+For a three-dimensional lattice, the :math:`x,y,z` coordinates of the lower-left
+coordinate need to be given and the pitch should also give dimensions for all
+three axes. For example, to make a 3x3x3 lattice where the bottom layer is
+universe ``u``, the middle layer is universe ``q`` and the top layer is universe
+``z`` would look like::
+
+  lat3d = openmc.RectLattice()
+  lat3d.lower_left = (-7.5, -7.5, -7.5)
+  lat3d.pitch = (5.0, 5.0, 5.0)
+  lat3d.universes = [
+      [[u, u, u],
+       [u, u, u],
+       [u, u, u]],
+      [[q, q, q],
+       [q, q, q],
+       [q, q, q]],
+      [[z, z, z],
+       [z, z, z]
+       [z, z, z]]]
+
+Again, using NumPy can make things easier::
+
+  lat3d.universes = np.empty((3, 3, 3), dtype=openmc.Universe)
+  lat3d.universes[0, ...] = u
+  lat3d.universes[1, ...] = q
+  lat3d.universes[2, ...] = z
+
+Finally, it's possible to specify that lattice positions that aren't normally
+without the bounds of the lattice be filled with an "outer" universe. This
+allows one to create a truly infinite lattice if desired. An outer universe is
+set with the :attr:`RectLattice.outer` attribute.
+
+Hexagonal Lattices
+------------------
+
+OpenMC also allows creationg of 2D and 3D hexagonal lattices. Creating a
+hexagonal lattice is similar to creating a rectangular lattice with a few
+differences:
+
+- The center of the lattice must be specified (:attr:`HexLattice.center`).
+- For a 2D hexagonal lattice, a single value for the pitch should be specified,
+  although it still needs to appear in a list. For a 3D hexagonal lattice, the
+  pitch in the radial and axial directions should be given.
+- As with rectangular lattices, the :attr:`HexLattice.outer` attribute will
```

Comment:
Another big difference is that it isn't as straightforward (or even possible?) to specify the universes using a NumPy array.